tests/test_analytics.py

```diff
@@ -70,6 +72,42 @@ def test_analytics_client_track_import(mock_datetime: MagicMock, mock_analytics:
     )
 
 
+@pytest.mark.skip(reason="Has a hardcoded timeout; could be too long for a unit test")
+def test_analytics_client_timeout():
+    def mock_urlopen(
+        req,
+        timeout,
+    ):
+        time.sleep(1)
+
+        @dataclass
+        class HTTPResponse:
+            status: int
+
+        return HTTPResponse(status=408)
+
+    urllib.request.urlopen = mock_urlopen
```

Comment:
Also for Python mocking, you'll want to use something like `patch()` (https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) which correctly patches the code only within the scope of the test

Reply:
The reason why I went the "manual" route here was because I didn't know how to use the `patch` decorator and have that mocked function throw an exception. I just recently found out that the way to do that is via `mocked_function.side_effect`.

Fixed according to your suggestion and pushed!